### PR TITLE
add support for linux/aarch64 for rust and csharp

### PR DESCRIPTION
**Pull request by xaiki**
from quick greps through the source this is the 2 placese where it's obvious it's broken, but there might be more.

### CODE REVIEW

File: clients/lsp-csharp.el [hunk 1]
```diff
@@ -69,26 +69,24 @@ Usually this is to be set in your .dir-locals.el on the project root directory."
...
-                (t "omnisharp-mono.zip")))
+          (pcase system-type
+            ('windows-nt
+             ; On Windows we're trying to avoid a crash starting 64bit .NET PE binaries in
+             ; Emacs by using x86 version of omnisharp-roslyn on older (<= 26.4) versions
+             ; of Emacs. See https://lists.nongnu.org/archive/html/bug-gnu-emacs/2017-06/msg00893.html"
+             (if (and (string-match "^x86_64-.*" system-configuration)
+                      (version<= "26.4" emacs-version))
+                 "omnisharp-win-x64.zip"
+               ((format "" ) "omnisharp-win-%s.zip"
```
Comment (yyoncho) [inline clients/lsp-csharp.el:80]: this `(format "")` does not look correct.

Comment (xaiki) [reply clients/lsp-csharp.el]: sorry, fixed.

**Comment by xaiki** [comment]
there is a bug in this code i can't immediately fix., let me get back to you

File: clients/lsp-csharp.el [hunk 2]
```diff
@@ -69,26 +69,24 @@ Usually this is to be set in your .dir-locals.el on the project root directory."
...
+          (pcase system-type
+            ('windows-nt
+             ; On Windows we're trying to avoid a crash starting 64bit .NET PE binaries in
+             ; Emacs by using x86 version of omnisharp-roslyn on older (<= 26.4) versions
+             ; of Emacs. See https://lists.nongnu.org/archive/html/bug-gnu-emacs/2017-06/msg00893.html"
+             (if (and (string-match "^x86_64-.*" system-configuration)
+                      (version<= "26.4" emacs-version))
+                 "omnisharp-win-x64.zip"
+               ((format "omnisharp-win-%s.zip"
+                        (lsp-resolve-value lsp--system-arch)))))
```
Comment (jcs090218) [inline clients/lsp-csharp.el:81]: I think you have 1 extra curly bracket here?

**Pull request closed without merging**